lib/stateTransition/AbstractStateTransition.js

```diff
@@ -190,7 +190,7 @@ class AbstractStateTransition {
    * @return {Object}
    */
   static translateJsonToObject(rawStateTransition) {
-    return rawStateTransition;
+    return { ...rawStateTransition };
```

Comment:
Don't we need a deep clone here? Just asking :))

Reply:
Yes we need. Right now we change rawStateTransition by link because of this.